container/image.bzl

```diff
@@ -124,84 +124,168 @@ def _image_config(
+            "-outputManifest",
+            "%s" % manifest.path,
         ]
 
-    ctx.actions.run(
-        executable = ctx.executable.create_image_config,
-        arguments = args,
-        inputs = inputs,
-        outputs = [config, manifest],
-        use_default_shell_env = True,
-        mnemonic = "ImageConfig",
-    )
+        if null_entrypoint:
+            args += ["-nullEntryPoint"]
+
+        if null_cmd:
+            args += ["-nullCmd"]
+
+        for x in entrypoint:
+            args += ["-entrypoint", "%s" % x]
+        for x in cmd:
+            args += ["-command", "%s" % x]
+        for x in ctx.attr.ports:
+            args += ["-ports", "%s" % x]
+        for x in ctx.attr.volumes:
+            args += ["-volumes", "%s" % x]
+
+        if creation_time:
+            args += ["-creationTime", "%s" % creation_time]
+        elif ctx.attr.stamp:
+            # If stamping is enabled, and the creation_time is not manually defined,
+            # default to '{BUILD_TIMESTAMP}'.
+            args += ["-creationTime", "{BUILD_TIMESTAMP}"]
+
+        for key, value in labels.items():
+            args += ["-labels", "%s" % "=".join([key, value])]
+
+        for key, value in env.items():
+            args += ["-env", "%s" % "=".join([
+                ctx.expand_make_variables("env", key, {}),
+                ctx.expand_make_variables("env", value, {}),
+            ])]
+
+        if ctx.attr.user:
+            args += ["-user", ctx.attr.user]
+        if workdir:
+            args += ["-workdir", workdir]
+
+        inputs = layer_names
+        for layer_name in layer_names:
+            args += ["-layerDigestFile", "@" + layer_name.path]
+
+        if ctx.attr.label_files:
+            inputs += ctx.files.label_files
+
+        if base_config:
+            args += ["-baseConfig", "%s" % base_config.path]
+            inputs += [base_config]
+
+        if base_manifest:
+            args += ["-baseManifest", "%s" % base_manifest.path]
+            inputs += [base_manifest]
+
+        if operating_system:
+            args += ["-operatingSystem", "%s" % operating_system]
+
+        if ctx.attr.stamp:
+            stamp_inputs = [ctx.info_file, ctx.version_file]
+            for f in stamp_inputs:
+                args += ["-stampInfoFile", "%s" % f.path]
+            inputs += stamp_inputs
+
+        if ctx.attr.launcher_args and not ctx.attr.launcher:
+            fail("launcher_args does nothing when launcher is not specified.", attr = "launcher_args")
+        if ctx.attr.launcher:
+            for x in ["/" + ctx.file.launcher.basename]:
+                args += ["-entrypointPrefix", "%s" % x]
+            args += ctx.attr.launcher_args
+        ctx.actions.run(
+            executable = ctx.executable.go_create_image_config,
+            arguments = args,
+            inputs = inputs,
+            outputs = [config, manifest],
+            use_default_shell_env = True,
+            mnemonic = "Go ImageConfig",
```

Comment:
nit: Convention is to not have spaces in the mnemonic. Suggest making this GoImageConfig